--- openmrn/src/utils/macros.h	2026-09-17 22:15:22.259329296 -0400
+++ patches/overrides/utils/macros.h	2026-09-17 22:17:58.137456918 -0400
@@ -102,6 +102,12 @@
 #include <esp32s3/rom/ets_sys.h>
 #elif defined(CONFIG_IDF_TARGET_ESP32C3)
 #include <esp32c3/rom/ets_sys.h>
+#elif defined(CONFIG_IDF_TARGET_ESP32C6)
+#include <esp32c6/rom/ets_sys.h>
+#elif defined(CONFIG_IDF_TARGET_ESP32H2)
+#include <esp32h2/rom/ets_sys.h>
+#elif defined(CONFIG_IDF_TARGET_ESP32C2)
+#include <esp32c2/rom/ets_sys.h>
 #else
 #error Unknown/Unsupported ESP32 variant.
 #endif // CONFIG_IDF_TARGET_ESP32
