--- openmrn/src/freertos_drivers/esp32/Esp32WiFiManager.cxx	2026-09-17 22:15:22.227329484 -0400
+++ patches/overrides/freertos_drivers/esp32/Esp32WiFiManager.cxx	2026-09-17 22:22:30.856945157 -0400
@@ -35,7 +35,7 @@
 // Ensure we only compile this code for ESP32 MCUs
 #ifdef ESP_PLATFORM
 
-#include "Esp32WiFiManager.hxx"
+#include "freertos_drivers/esp32/Esp32WiFiManager.hxx"
 #include "openlcb/SimpleStack.hxx"
 #include "openlcb/TcpDefs.hxx"
 #include "os/MDNS.hxx"
@@ -61,6 +61,8 @@
 #include <esp32s2/rom/crc.h>
 #elif defined(CONFIG_IDF_TARGET_ESP32C3)
 #include <esp32c3/rom/crc.h>
+#elif defined(CONFIG_IDF_TARGET_ESP32C6)
+#include <esp32c6/rom/crc.h>
 #elif defined(CONFIG_IDF_TARGET_ESP32S3)
 #include <esp32s3/rom/crc.h>
 #else // default to ESP32
